### ISSUE TITLE
Strange error statement: Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n

### ISSUE THREAD

**Issue by nishakm** [nexB/scancode-toolkit#1873]
### Description
I ran a scan on what looks like a minimized Debian filesystem (I got it from a Docker container):
`scancode -lpcu --quiet --json - /path/to/filesystem`
Before it spits out the giant json object it gives this message:
`Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n`

When running without `--quiet` I get this message:
```
Scanning done.
Some files failed to scan properly:
Path: contents/usr/share/doc/gpgv/copyright
Path: contents/usr/share/doc/libgcrypt20/copyright
```
I don't see any issue with these two files.

When running the scan on just these two files I don't get the above "some characters could not be decoded" but it does give a scan error:
```
"ERROR: for scanner: licenses:\nERROR: Unknown error:\nTraceback (most recent call last):\n  File \"/home/nisha/scancodeenv/lib/python3.6/site-packages/scancode/interrupt.py\", line 92, in interruptible\n    return NO_ERROR, func(*(args or ()), **(kwargs or {}))\n  File \"/home/nisha/scancodeenv/lib/python3.6/site-packages/scancode/api.py\", line 211, in get_licenses\n    _licenses_data_from_match(match, matched_text, license_url_template)\n  File \"/home/nisha/scancodeenv/lib/python3.6/site-packages/scancode/api.py\", line 234, in _licenses_data_from_match\n    result['key'] = lic.key\nAttributeError: 'NoneType' object has no attribute 'key'\n"]}
```

### System configuration

System: Ubuntu 18.04
Scancode version: 3.1.1
Downloaded using pip install [scancode-toolkit](url)
Python version: 3.7

**Comment by nishakm** [comment]
Copyright files attached:
[gpgv_copyright.txt](https://github.com/nexB/scancode-toolkit/files/4035782/gpgv_copyright.txt)
[libgcrypto20_copyright.txt](https://github.com/nexB/scancode-toolkit/files/4035783/libgcrypto20_copyright.txt)

**Comment by pombredanne** [comment]
@nishakm Thanks! that's a bug alright but I cannot reproduce it for now on the head of develop. 
I will check it more and get a fix in tomorrow if it needs one.  But can you tell which version of Python you run?

**Comment by nishakm** [comment]
@pombredanne Of course! It's Python 3.7 in a virtual environment. I'll add that to the issue description.

**Comment by nishakm** [comment]
@pombredanne Confirmed - I don't see the error when running scancode at the the head of develop. I believe 3.1.1 is the latest pip release?

**Comment by pombredanne** [comment]
@nishakm re:

>  I don't see the error when running scancode at the the head of develop. I believe 3.1.1 is the latest pip release?

Thank you for testing! yes, 3.1.1 is the latest @ PyPI and we are working on a new release ATM.

**Comment by pombredanne** [comment]
This is fixed in develop, so closing.
Release tag coming soon at a repo near you!

**Comment by pombredanne** [comment]
This is showing up again in the latest Python version, so I am reopening,

**Comment by sohang3112** [comment]
@pombredanne I recently had a similar issue in Docker (not in `scancode`) which had this same error: *Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.* It turned out to be because the website was sending `gzip` compressed content (`Content-Encoding=gzip` in response headers), but the Docker image did not support `gzip`. I solved it by removing `"Accept-Encoding": "*"` request header while requesting from client code (specifying `"Accept-Encoding": "identity"` should also work, i.e., get uncompressed content from website).

I don't know if this will solve the error here or not, but it can be tried.

**Comment by pombredanne** [comment]
@sohang3112 re:

> but the Docker image did not support gzip.

Is this a recent Fedora?

**Comment by sohang3112** [comment]
> Is this a recent Fedora?

@pombredanne No, I used `python:3.8` Docker image (which is Debian-based). Not sure if `gzip` is not installed in it, or if `gzip` was incorrectly configured.

**Comment by pombredanne** [comment]
BTW, @sohang3112  here the warning comes from https://git.launchpad.net/beautifulsoup/tree/bs4/dammit.py#n616
There are ways to filter spurious log messages in Python.
You are using ScanCode with Docker, right?

**Comment by sohang3112** [comment]
@pombredanne No, I am NOT using `scancode` at all - doing something unrelated and came across this issue via Google. Just commented in case the same error is happening here.
